### ISSUE TITLE
Build all of my targets (iOS) simultaneously, with one lane 

### ISSUE THREAD

**Issue by jpghelfi** [fastlane/fastlane#11954]
### New Issue Checklist (NOT AN ISSUE)

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Is there a way to run multiple lanes at the same time, if the computer has the enough power 

<run>
  <pre>fastlane all
</pre>
</run>

<fastfile>
  <pre>
lane :all do
    lane betaTarget1
    lane betaTarbet2
end
</pre>
</fastfile>

**Comment by mgrebenets** [comment]
@jpghelfi Do you mean in parallel, i.e. run them concurrently?

I guess you can, but not sure if you can do it with single fastlane invocation.
Fastlane runs the lanes in Fastfile "top to bottom" as if it was just a Ruby script.

Even if you open multiple shell sessions in same directory and run different lanes in each shell session, I'm afraid those parallel invocations may be writing/overwriting each other's files in the same directory or _maybe_ it all just works fine.

You could create multiple copies of your project folder and run fastlane lane in each folder in parallel, then there's no race condition and each lane is properly sandboxed. Not sure if that would work for you, because I'd imagine you want to run all builds as part of one CI job and collect all artifacts in the end?

You could also use some Ruby gem (if there's any) which can create subprocesses and synchronize them all in Ruby code...

Anyhow, short answer, Fastlane doesn't have such a parallelization feature out of the box afaik.

If the main concern is to utilize as much CPU power as possible, then I wish Apple would just improve `xcodebuild` so it is better with parallelizing it's build tasks :/
